File: backend/api/export.py
from __future__ import annotations

from fastapi import APIRouter

from backend.api.errors import NotFound
from backend.api.schemas import ExportJob, ExportRequest, JobOut
from backend.config import get_settings
from backend.lib.jobs.ejecutor import submit
from backend.lib.jobs.registro import JobState, registry
from backend.services.export import ExportService

router = APIRouter(prefix="/api", tags=["export"])


def _service() -> ExportService:
    return ExportService(get_settings())


def _out(job: JobState) -> JobOut:
    return JobOut(
        jobId=job.job_id,
        status=job.status,
        progress=job.progress,
        result=job.result,
        error=job.error,
    )


@router.get("/games/{game_id}/export-options")
def export_options(game_id: str) -> dict[str, list[dict[str, object]]]:
    return _service().options(game_id)


@router.post("/export")
def create_export(payload: ExportRequest) -> ExportJob:
    job = submit(_service().export_job(payload.gameId, payload.incluir))
    return ExportJob(runId=job.job_id)


@router.get("/export/{run_id}")
def get_export(run_id: str) -> JobOut:
    job = registry.get(run_id)
    if job is None:
        raise NotFound(f"Export no encontrado: {run_id}")
    return _out(job)

File: frontend/src/lib/api/export.ts
import { fetchJson } from './client';

export interface ExportOption {
  key: string;
  label: string;
  required: boolean;
  disponible: boolean;
  bytes: number;
}

export interface ExportOptions {
  obligatorio: ExportOption[];
  opcional: ExportOption[];
}

export interface ExportJob {
  runId: string;
}

export interface ExportResult {
  file: string;
  bytes: number;
  incluye: string[];
  verificado: Record<string, unknown>;
}

export interface JobOut {
  jobId: string;
  status: string;
  progress: number;
  result: ExportResult | null;
  error: string | null;
}

export function getExportOptions(gameId: string) {
  return fetchJson<ExportOptions>(`/games/${gameId}/export-options`);
}

export function createExport(gameId: string, incluir: string[]) {
  return fetchJson<ExportJob>('/export', {
    method: 'POST',
    body: JSON.stringify({ gameId, incluir }),
  });
}

export function getExportStatus(runId: string) {
  return fetchJson<JobOut>(`/export/${runId}`);
}
